test/manipulation.test.js

```diff
@@ -1598,11 +1613,19 @@ describe('manipulation', function() {
         id: created.id,
         title: 'Draft',
         content: 'a content',
+        throwingSetter: undefined,
```

Comment:
I am afraid this change breaks tests in SQL connectors, where properties with no values are set to `null`, not `undefined`.

Unfortunately, CI results from downstream tests are not publicly available, I am cross-posting the test failure reported by our PostgreSQL connector:

```

  1) juggler-v4
       manipulation
         replaceById
           correctly coerces the PK value:

      AssertionError: expected Object { content: 'a content', id: 1, throwingSetter: null, title: 'Draft' } to equal Object {
  content: 'a content',
  id: 1,
  throwingSetter: undefined,
  title: 'Draft'
} (at throwingSetter, A has null and B has undefined)
      + expected - actual

       {
         "content": "a content"
         "id": 1
      -  "throwingSetter": [null]
      +  "throwingSetter": [undefined]
         "title": "Draft"
       }
      
      at Assertion.fail (deps/juggler-v3/node_modules/should/lib/assertion.js:92:17)
      at Assertion.value (deps/juggler-v3/node_modules/should/lib/assertion.js:164:19)
      at Context.it (node_modules/loopback-datasource-juggler/test/manipulation.test.js:1612:31)
      at <anonymous>
      at process._tickCallback (internal/process/next_tick.js:189:7)
```

I think the easier fix is to always use `null` for `throwingSetter`, even when it's not set to any value. I am not sure what's the best approach for that - you can try to add `default: null` flag to the property definition, or maybe implement a getter function that will always return `null`?

The last resort is to enable `persistUndefinedAsNull` in model settings, but I am concerned that such change can affect other tests in unwanted ways.

Reply:
I added a default **null** type to the `throwingSetter` value. Could you please check now?

Reply:
Much better! SQL tests are passing now. The Cloudant error seems to be caused by a problem in the CI infrastructure, not by your changes. I'll re-run the build to verify.